synapse/federation/federation_server.py

```diff
@@ -99,10 +99,15 @@ def __init__(self, hs):
         super().__init__(hs)
 
         self.auth = hs.get_auth()
-        self.handler = hs.get_handlers().federation_handler
+        self.handler = hs.get_federation_handler()
         self.state = hs.get_state_handler()
 
         self.device_handler = hs.get_device_handler()
+
+        # Ensure the following handlers are loaded since they register callbacks
+        # with FederationHandlerRegistry.
+        hs.get_directory_handler()
```

Comment:
This is necessary because previously `get_handlers()` would instantiate the directory handler (and a few other handlers). Only the directory handler called `register_edu_handler` or `register_query_handler`.